### DISCUSSION TITLE
Building a case for OS - impact/risks and investing talent

### DISCUSSION THREAD

**Discussion by epijim** [rinpharma/positconf-roundtables-2023#6] [Ideas]
Where are we today across different companies in terms of getting buy in for building tools in an open setting. Are we capturing the return on investment (e.g. contributions back), and if so how?

What gaps are present in terms of getting buy in to invest in  moving tool development into an open setting, or transitioning off legacy propriety tools (e.g. macros, GUI tools like spotfire) to new open source tools (e.g. shiny, pharmaverse R packages, HTA R packages, etc)

**Comment by dgkf-roche** [comment]
To tack a related idea onto this topic, I think that there is a specific risk of companies cutting resources after the open source option is available - outsourcing development effort to an open source team of volunteers or other pharmas.

I think the `pharmaverse` has a bit more guidance around how companies should be donating resources and their expectations around contribution from industry users. I'd be interested to learn more about that as the R Validation Hub has aspirations of hosting a package repository that comes closer to satisfying our regulatory needs. My fear is that organizations will cut back their teams instead of shifting focus toward contributing to a collective solution. 

I would feel more comfortable knowing that the messaging was consistent across the industry. My primary goal is not resource reduction - it's access to tooling, consistency and transparency. Perhaps consistency means we can be more efficient with our collective headcount overall, but I see that as a long-term side effect, not an immediate deliverable.

**Reply by kamilsi** [reply]
@dgkf-roche, you bring up an important concern about organizations potentially reallocating resources away from their in-house development teams once an open-source alternative is available. I share your concern but offer a slightly different perspective.

In my experience, layoffs in IT departments are not as common as one might think, especially in sectors like healthcare and pharmaceuticals where the demand for technological solutions is ever-growing. Solving one problem with an open-source tool doesn't usually mean there's nothing left to do; instead, it frees up valuable time and brainpower to tackle new, perhaps even more complex, challenges.

Moreover, there's a strong appeal for open-source projects within the medical sector, driven by what could be termed the "hacker ethos". Richard Stallman encapsulated this beautifully in his [Code of Ethics for Hackers](https://www.expertbase.org/a123-The-Hacker's-Code-of-Ethics!): "No problem should ever have to be solved twice. Creative brains are a valuable, limited resource. They shouldn't be wasted on re-inventing the wheel when there are so many fascinating new problems waiting out there."

So, while the potential for reduced internal resources is a valid concern, I remain optimistic that the shift toward open-source could actually lead to an increase in problem-solving across the board. I'm keeping my fingers crossed that this shift enables us to tackle more challenges than ever before.

**Comment by minhonglee** [comment]
I think there are a couple of barriers to get more participation/contribution to open source projects:  whether there is benefit to the company itself and whether there's any need to share sponsor specific proprietary data.  

Sponsor can support staff development and interest in pursuing open source software development even though it may not directly contribute to the benefit of their organization but it would obviously be much easier if the participants are contributing both to the project but also to the company's goals as well.  I think the focus has to be on the benefits of the open source project itself and how broadly applicable or relevant it is to industry.   Project's that are too broad in scope and may be trying to do too much or not sufficiently focused on specific deliverables may be difficult to get support versus any effort that has specific problems or challenges that they're trying to solve.   I think the effort by the R Validation Hub for a package repository falls into the real world problem category that many companies are challenged with and could be something that gets broad support if appropriately scoped and communicated.  

Another challenge is that some efforts that address real world issues that companies are facing may require the use of proprietary data.  Unfortunately, because companies have not really worked together very much in this way, we will probably be more risk adverse in our willingness to share data.    Hopefully, time will improve a company's willingness to support open source development.  The more experience we have and the more success stories we can point to, we have better opportunities to collaborate.  

I think one thing that does support more participation rather than less is that one of the drivers for moving to open-source is maintaining, developing, and attracting talent, and their desire to collaborate with their industry colleagues.

**Comment by kamilsi** [comment]
@epijim, your question about the buy-in for open-source tool development is timely and essential. At my company, one of the early barriers we faced was understanding how GPL licenses would mesh with our proprietary codebase. We even devoted a talk to this topic at a local community meetup, discussing the complexities around GPL licenses in commercial settings (you can find the slides [here](https://github.com/PyStok/PyStok-43/blob/master/Licencje%20GPL%20i%20dzia%C5%82alno%C5%9B%C4%87%20komercyjna%20-%20jakie%20mamy%20mo%C5%BCliwo%C5%9Bci.pdf)).

There is a legal landscape that organizations need to navigate when adopting open-source technologies, especially when these technologies will be integrated into commercial products or services. Misunderstandings or misinterpretations of licenses can expose companies to significant legal risks. Hence, legal clarity is often one of the foremost gaps that need to be filled for broader acceptance and investment in open-source tool development. This could be an area where organizations in the pharmaverse could pool resources to provide robust guidance.
